.github/workflows/test.yaml

```diff
@@ -213,3 +213,29 @@ jobs:
           PROPTEST_CASES: "10"
           MIRIFLAGS: "-Zmiri-disable-isolation"
         run: cargo miri test --all-features
+
+  thread_sanitizer:
+    name: Thread Sanitizer checks
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ github.event.repository.name }}-${{ runner.os }}-cargo-thread_sanitizer-v2
+    - name: Install latest nightly
+      uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly
+          override: true
+          components: rust-src
+    - name: Run thread sanitizer
+      run: |
+        export RUSTDOCFLAGS='-Zsanitizer=thread'
+        export RUSTFLAGS='-Zsanitizer=thread'
+        for _ in $(seq 1 10); do cargo +nightly test -Z build-std --target --target $(uname -m)-unknown-linux-gnu; done
```

Comment:
What is the for cycle for in here? If it really serves some particular purpose, a comment explaining why such weirdness is necessary would be nice.

Reply:
Thread sanitizer isn't guaranteed to catch all data races, it can only catch it if the data race happens when running the program.

Running the tests multiple times increase the chances that data races are found by thread sanitizer.

Reply:
I've added comment to explain the loop used here.

Reply:
Well, that's true for most of the tests in around this crate. Mostly, that's why the stress tests do multiple iterations internally.

But I guess it's OK.